### PR TITLE
Etcm 52 make sure mantis works on windows

### PR DESCRIPTION
**Pull request by kapke**
# Description

This PR fixes issues found when trying to run Mantis on Windows, that is:
  - generating proper `.bat` files
  - fixing the way `SecureRandom` instance is obtained
  - KeyStore tests which were too UNIX-oriented
  - Update RocksDB to latest version, so Ethereum fork doesn't need to be used

# Testing

All tests, `sbt dist` and syncing from both sbt and dist package should work

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -100,8 +97,11 @@ scalacOptions in (Compile, console) ~= (_.filterNot(Set(
   "-Ywarn-unused-import",
   "-Xfatal-warnings"
 )))
+Global / onChangedBuildSource := ReloadOnSourceChanges
+
+Test / parallelExecution := false
 
-parallelExecution in Test := false
+ThisBuild / turbo := true
```
Comment (ntallar) [inline build.sbt:104]: Why should we enable this? From the description:

>sbt 1.3.0 adds “turbo” mode that enables experimental or advanced features that might require some debugging by the build user when it doesn’t work. 

The advanced features might be useful, but I'm a bit cautious on opening the door to any experimental features

File: build.sbt [hunk 2]
```diff
@@ -119,11 +119,12 @@ unmanagedResourceDirectories in Compile += baseDirectory.value / "src" / "main"
 (scalastyleConfig in Test) := baseDirectory.value / "scalastyle-test-config.xml"
 scalastyleSources in Test ++= {(unmanagedSourceDirectories in Integration).value}
 
+// Packaging
 mainClass in Compile := Some("io.iohk.ethereum.App")
-
+discoveredMainClasses in Compile := Seq("io.iohk.ethereum.mallet.main.Mallet")
```
Comment (ntallar) [inline build.sbt:124]: This doesn't seem to work on my machine:

```
 % ./bin/mallet
Error: Could not find or load main class io.iohk.ethereum.mallet.main.Mallet
```

File: src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala [hunk 3]
```diff
@@ -193,10 +193,10 @@ class RocksDbDataSource(
 
       val tableCfg = new BlockBasedTableConfig()
         .setBlockSize(blockSize)
-        .setBlockCacheSize(blockCacheSize)
+        .setBlockCache(new ClockCache(blockCacheSize))
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala:196]: Just to understand the decision, why ClockCache and not LRUCache? From what I read LRUCache has more overhead

File: src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala [hunk 4]
```diff
@@ -510,7 +510,7 @@ trait GenesisDataLoaderBuilder {
 
 trait SecureRandomBuilder {
   lazy val secureRandom: SecureRandom =
-    Config.secureRandomAlgo.map(SecureRandom.getInstance).getOrElse(new SecureRandom())
+    Config.secureRandomAlgo.flatMap(name => Try(SecureRandom.getInstance(name)).toOption).getOrElse(new SecureRandom())
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala:513]: Doesn't this hide the fact that the selected algorithm is available and other one is instead being used?

**Review by ntallar — CHANGES_REQUESTED** [review]
Both running Mantis from `sbt dist` or `sbt run` worked on my machine! (it's a linux machine)

File: build.sbt (hunk 1 above)
Comment (kapke) [reply build.sbt:104]: One of most important things it changes is classloader layering. Enabling turbo mode didn't broke anything so my preference is to keep it enabled and disable it only when it does break something for us.

File: build.sbt (hunk 2 above)
Comment (kapke) [reply build.sbt:124]: Yup, once I realaized that this probably needs to be adjusted, @mirkoAlic confirmed for me that mallet can be removed so there will be a separate PR soon I think

File: src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala (hunk 3 above)
Comment (kapke) [reply src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala:196]: Exactly this - documentation says they're very similar but `ClockCache` can perform better in certain scenarios so it seem to be better option for start.

File: src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala (hunk 4 above)
Comment (kapke) [reply src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala:513]: If I add logging about algorithm non available, will it be sufficient?

**Review by KonradStaniec — APPROVED** [review]
I had run mantis on macos, from pre-exsiting database and it seems to work fine. @kapke did you check it on windows ?

**Comment by kapke** [comment]
I did, whole PR was prepared on Windows machine

File: src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala (hunk 4 above)
Comment (ntallar) [reply src/main/scala/io/iohk/ethereum/nodebuilder/NodeBuilder.scala:513]: Yes, given our current handling of config errors I think that with logging with error will be enough here

**Review by ntallar — APPROVED** [review]
LGTM!
```
 .\\            //.
. \ \          / /.
.\  ,\     /` /,.-
 -.   \  /'/ /  .
 ` -   `-'  \  -
   '.       /.\`
      -    .-
      :`//.'
      .`.'
      .' BP 
```

**Pull request merged**